src/request.rs

```diff
@@ -80,11 +85,30 @@ impl Request {
                 Header::ContentLength => {
                     // check if value is usize:
                     self.content_length = Some(self.headers.len());
+                    match usize::from_str(value.as_str()) {
+                        Ok(size) => {
+                            if body_kind != BodyKind::Chunked {
+                                body_kind = BodyKind::Fixed(size);
+                            }
+                        },
+                        _ => {},
+                    }
                 },
+                Header::TransferEncoding => {
+                    self.transfer_encoding = Some(self.headers.len());
+                    match value.split(|c| c == ',').last() {
+                        Some("chunked") => {
+                            body_kind = BodyKind::Chunked;
```

Comment:
If content-length is also set we should also add `Connection: close` because we don't count bytes of chunked and don't compare it with Content-Length I think.